xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp

```diff
@@ -1042,6 +1042,16 @@ unsigned CDVDVideoCodecAndroidMediaCodec::GetAllowedReferences()
   return 4;
 }
 
+float CDVDVideoCodecAndroidMediaCodec::GetFramerate() const
+{
+  int fpsRate = m_hints.fpsrate;
+  if (m_hints.fieldOrder > AV_FIELD_PROGRESSIVE)
+    fpsRate *= 2;
```

Comment:
This mustn't cunfusing you badly, this is a PR and open for discussion.
Seems that you have lot of knowledge about different codecs and their behaviour, pls. provide information about how it should be solved.

Reply:
The info is wrong in many cases anyway. So it does not seem to be crucial.

The source is 25fps, no matter whether content is truly interlaced or from progressive source or what the deinterlacer does.

I did some testing with 2 DVDs on my BRAVIA Android TV.

(1) The Mummy: Tomb of the Dragon Emperor: progressive 576i25 (50 fields per second with always 2 fields originating from the same point in time)
(2) Rocky: truly interlaced 576i25 (50 fields per second with each originating from a different point in time)

(1) can be deinterlaced with a simple WEAVE, putting 2 fields together to build one single frame, resulting in 25fps. (2) is either deinterlaced to 25fps (half rate) or 50fps (full rate).

I played those DVDs in Kodi through Android MediaCodec and ffmpeg (MediaCodec disabled). Both output 25fps for both DVDs, therefore performing half rate deinterlacing for truly interlaced material.

ffmpeg in Kodi is configured to perform half rate deinterlacing (Yadif: 0,-1,1) on Android (I think @fritsch changed that because of the many slow boxes and TVs). It looks like deinterlacer can't be switched to full rate (Yadif: 1,-1,1) in video settings (bug?).

I would leave the info at 25fps because it is the correct one from a source point of view. Either truly interlaced or interlaced from progressive source, 25fps is correct in any case. As for refresh rate switching, I would always switch to 50Hz in case of PAL if a respective display mode exists, even if you have 25fps at the renderer...

Reply:
[CVideoPlayerVideo::m_fFrameRate](https://github.com/xbmc/xbmc/blob/770c691bfeea88d561b39d2e38348ac434db3adc/xbmc/cores/VideoPlayer/VideoPlayerVideo.h#L111) might be the right figure to look at.

Reply:
Thx @CiNcH83, can you provide a some second snippet of "The Mummy" pls ?

Reply:
Sure...

[The Mummy: Tomb of the Dragon Emperor: progressive 576i25](https://www.dropbox.com/s/lwi7szljq6jx9cv/576i25_progressive.mkv?dl=1)
```
Format                                   : MPEG Video
Format version                           : Version 2
Format profile                           : Main@Main
Width                                    : 720 pixels
Height                                   : 576 pixels
Frame rate                               : 25.000 FPS
Standard                                 : PAL
Scan type                                : Progressive
Scan order                               : Top Field First
```

[Rocky: interlaced 576i25](https://www.dropbox.com/s/nisjgs74eacvu5o/576i25_interlaced.mkv?dl=1)
```
Format                                   : MPEG Video
Format version                           : Version 2
Format profile                           : Main@Main
Width                                    : 720 pixels
Height                                   : 576 pixels
Frame rate                               : 25.000 FPS
Standard                                 : PAL
Scan type                                : Interlaced
Scan order                               : Top Field First
```